src/wallet/wallet.cpp

```diff
@@ -3183,8 +3201,9 @@ bool CWallet::TopUpKeyPool(unsigned int kpSize)
                 nEnd = std::max(nEnd, *(setExternalKeyPool.rbegin()) + 1);
             }
 
-            if (!walletdb.WritePool(nEnd, CKeyPool(GenerateNewKey(internal), internal)))
+            if (!walletdb.WritePool(nEnd, CKeyPool(GenerateNewKey(walletdb, internal), internal))) {
                 throw std::runtime_error(std::string(__func__) + ": writing generated key failed");
+            }
 
             if (internal) {
                 setInternalKeyPool.insert(nEnd);
```

Comment:
If we increase the default keypool size, the `LogPrintf("keypool added key %d` message below shouldn't be logged for every key. Or at least moved to a debug class.